2-Draft-Accepted/RFCxxxx-DSC-for-PowerShell7.md

```diff
@@ -0,0 +1,318 @@
+---
+RFC: RFCxxxx
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 1.0
+Area: Desired State Configuration
+Comments Due: Nov 22, 2020
+Plan to implement: Yes
+---
+
+# DSC for PowerShell 7
+
+[Desired State Configuration](https://docs.microsoft.com/powershell/scripting/dsc/overview/overview)
+(DSC) is a platform in PowerShell enabling configuration as code.
+
+## Background
+
+DSC contains many components including:
+
+ - The `configuration` keyword that is part of the PowerShell engine
+ - Compilation to a configuration MOF file
+ - DSC resources (those that ship in Windows and those on PowerShellGallery.com)
+ - Local Configuration Manager in Windows and maintained as open-source for Linux
+ - Pull Server support
+ - PSDesiredStateConfiguration module.
+
+DSC ships as a component of Windows and as an open source project for Linux.
+It is also integrated in to cloud-based server management services in multiple
+public clouds including Microsoft Azure and Amazon AWS.
+
+The versions of DSC that shipped in Windows PowerShell are tightly coupled with
+[Windows Management Instrumentation (WMI)](https://docs.microsoft.com/windows/win32/wmisdk/wmi-start-page)
+on Windows and
+[Open Management Interface](https://www.dmtf.org/content/open-management-interface-omi-%E2%80%93-open-source-implementation-dmtf-cimwbem-standards)
+(OMI) on Linux.
+The *MI libraries are used to host the Local Configuration Manager (LCM) and APIs
+are used to parse information store in Managed Object Framework (MOF) file format,
+which is used for both configuration files and schema files.
+
+### Challenges with the current versions of DSC
+
+Very few community authored DSC resources exist for Linux because C++ is the only
```

Comment:
Isn't Python also supported? Or did we never officially support that?

Reply:
Michael asked the same question.  I thought it was supported, but not aware of any resources built on it.

Reply:
From a convo with Norberto a couple years ago, I thought that _at a minimum_ the nxFileLine resource was built with Python.

Reply:
Ahhh, searched my email.  It was a Python USE.  The resource was still built in CPP.